examples/heat-equation/heat-equation.cpp

```diff
@@ -0,0 +1,209 @@
+{
+    cv::Size videosize{n, n};
+    auto output =
+        std::make_pair(cv::VideoWriter{}, cv::Mat{videosize, CV_8UC3});
+    auto fourcc = cv::VideoWriter::fourcc('a', 'v', 'c', '1');
+    output.first.open("heat.mp4", fourcc, fps, videosize);
+    return output;
+}
+
+
+// Write the current frame to video output using the above color mapping
+void output_timestep(std::pair<cv::VideoWriter, cv::Mat> &output, int n,
+                     const double *data)
+{
+    for (int i = 0; i < n; i++) {
+        auto row = output.second.ptr(i);
+        for (int j = 0; j < n; j++) {
+            set_val(&row[3 * j], data[i * n + j]);
+        }
+    }
+    output.first.write(output.second);
+}
+
+
+int main(int argc, char *argv[])
+{
+    using mtx = gko::matrix::Csr<>;
+    using vec = gko::matrix::Dense<>;
+
+    // Problem parameters:
+    // simulation length
+    auto t0 = 5.0;
+    // diffusion factor
+    auto diffusion = 0.0005;
+    // scaling factor for heat source
+    auto source_scale = 2.5;
+    // Simulation parameters:
+    // grid size for inner points
+    auto n = 256;
+    // number of simulation steps per second
+    auto steps_per_sec = 500;
+    // number of video frames per second
+    auto fps = 25;
+    // number of grid points
+    auto n2 = n * n;
+    // grid point distance
+    auto h = 1.0 / n;
+    auto h2 = h * h;
+    // time step size for the simulation
+    auto tau = 1.0 / steps_per_sec;
+
+    // create an OpenMP executor
+    auto exec = gko::OmpExecutor::create();
+    // load heat source and initial state vectors
+    std::ifstream source_stream("data/source.mtx");
+    std::ifstream initial_stream("data/initial.mtx");
+    auto source = gko::read<vec>(source_stream, exec);
+    auto in_vector = gko::read<vec>(initial_stream, exec);
+    // create output vector with initial guess for
+    auto out_vector = in_vector->clone();
+    // create scalar for source update
+    auto tau_source_scalar = gko::initialize<vec>({source_scale * tau}, exec);
+    // create stencil matrix as shared_ptr for solver
+    auto stencil_matrix = gko::share(mtx::create(exec));
+    // assemble matrix
+    gko::matrix_data<> mtx_data{gko::dim<2>(n2, n2)};
+    for (int i = 0; i < n; i++) {
+        for (int j = 0; j < n; j++) {
+            auto c = i * n + j;
+            auto c_val = diffusion * tau * 4.0 / h2 + 1.0;
+            auto off_val = -diffusion * tau / h2;
+            // for each grid point: insert 5 stencil points
+            // with Dirichlet boundary conditions, i.e. with zero boundary value
+            // clang-format off
+            if (i > 0) mtx_data.nonzeros.emplace_back(c, c - n, off_val);
+            if (j > 0) mtx_data.nonzeros.emplace_back(c, c - 1, off_val);
+            mtx_data.nonzeros.emplace_back(c, c, c_val);
+            if (j < n - 1) mtx_data.nonzeros.emplace_back(c, c + 1, off_val);
+            if (i < n - 1) mtx_data.nonzeros.emplace_back(c, c + n, off_val);
```

Comment:
we have the brackets rule for control flow in [control-flow-constructs](https://github.com/ginkgo-project/ginkgo/wiki/Contributing-guidelines#control-flow-constructs).
Do you think we can allow something for readability?

Reply:
This was mainly meant for the poster, where vertical space seems to be more limited, iirc. I have no issue with reformatting this at all.